source/E2E.Tests/Services/BesiegerCamps/BesiegerCampCollectionTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿using E2E.Tests.Environment;
+using E2E.Tests.Environment.Instance;
+using E2E.Tests.Util;
+using GameInterface.Services.BesiegerCamps.Patches;
+using HarmonyLib;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Siege;
+using Xunit.Abstractions;
+
+namespace E2E.Tests.Services.BesiegerCamps
+{
+    public class BesiegerCampCollectionTests : IDisposable
+    {
+        private List<MethodBase> disabledMethods = new();
+
+        private E2ETestEnvironment TestEnvironment { get; }
+
+        private EnvironmentInstance Server => TestEnvironment.Server;
+        private IEnumerable<EnvironmentInstance> Clients => TestEnvironment.Clients;
+
+        public BesiegerCampCollectionTests(ITestOutputHelper output)
+        {
+            TestEnvironment = new E2ETestEnvironment(output);
+            DisableMethods();
+        }
+
+        public void Dispose()
+        {
+            TestEnvironment.Dispose();
+        }
+
+        private void DisableMethods()
+        {
+            disabledMethods = new List<MethodBase> {
+                AccessTools.Method(typeof (MobileParty), nameof(MobileParty.OnPartyJoinedSiegeInternal)),
+                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.SetSiegeCampPartyPosition)),
+                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.InitializeSiegeEventSide))
+            };
+        }
+
+        private T ServerCreateObject<T>(out string objectId)
+        {
+            string? id = null;
+            T? obj = default;
+
+            Server.Call(() =>
+            {
+                obj = GameObjectCreator.CreateInitializedObject<T>();
+                Assert.True(Server.ObjectManager.TryGetId(obj, out id));
+            }, disabledMethods);
+
+            objectId = id!;
+            return obj ?? throw new InvalidOperationException("Failed to create object.");
+        }
```

Comment:
This already exists in the [E2ETestEnvironment class](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/E2E.Tests/Environment/E2ETestEnvironment.cs#L85)

Use `TestEnvironment.CreateRegisteredObject<T>()` instead

Reply:
Could It have an optional arg of `List<MethodBase> `?, because I need a way to pass the disabled methods.

Reply:
Is it under a new name or something?, can't find anything

Reply:
Did you pull from remote?

Reply:
I might have erased your commit from history when force pushing in order to rebase onto development, could you push it again?

Reply:
I had a merge conflict and never pushed, it's in the branch now. My apologies

Reply:
np, using it now